File: jiwon/개인정보수집유효기간.java
import java.util.*;
import java.util.stream.Stream;
class Solution {
    public int[] solution(String today, String[] terms, String[] privacies) {
        ArrayList<Integer> answer = new ArrayList<>();
        //1. 약관일을 담을 배열 생성
        int [] termsDay = new int[26];
        for(String term : terms){
            int index = term.split(" ")[0].charAt(0) - 'A';
            int day = Integer.parseInt(term.split(" ")[1]) * 28;
            termsDay[index] = day;
        }
        // System.out.println(Arrays.toString(termsDay));

        //2. 현재 날짜일 계산

        int Today = Integer.parseInt(today.split("\\.")[0])*12*28
                + Integer.parseInt(today.split("\\.")[1])*28
                + Integer.parseInt(today.split("\\.")[2]);

        // System.out.println(Today);

        //3. for privacies 로 폐기 여부 확인
        for(int i=0; i<privacies.length; i++){
            String [] temp = privacies[i].split(" ");
            int privacyDay = Integer.parseInt(temp[0].split("\\.")[0])*12*28
                    + Integer.parseInt(temp[0].split("\\.")[1])*28
                    + Integer.parseInt(temp[0].split("\\.")[2]);
            if( privacyDay + termsDay [ temp[1].charAt(0)-'A' ] - Today <= 0){
                answer.add(i+1);
            }
        }


        return answer.stream().mapToInt(Integer::intValue).toArray();
    }
}

File: joon/src/programmers/kakao2023blind/개인정보 수집 유효기간.py
def solution(today, terms, privacies):
    t = {}
    d = today.split('.')
    d = [int(i) for i in d]
    answer = []

    for i in range(len(terms)):
        t[terms[i][0]] = terms[i][2:]

    for i in range(len(privacies)):
        date = privacies[i][:-2].split('.')

        if bigyo(d, date, t[privacies[i][-1]]):
            answer.append(i + 1)

    return answer

def bigyo(d, date, m):
    date = [int(i) for i in date]

    date[1] += int(m)
    while date[1] > 12:
        date[1] -= 12
        date[0] += 1

    if date[0] < d[0]:
        return True
    elif date[0] == d[0] and date[1] < d[1]:
        return True
    elif date[0] == d[0] and date[1] == d[1] and date[2] <= d[2]:
        return True
    return False